cpp/include/cudf/join.hpp

```diff
@@ -438,6 +446,64 @@ class hash_join {
   const std::unique_ptr<impl_type const> _impl;
 };
 
+/**
+ * @brief Unique hash join that builds hash table in creation and probes results in subsequent
+ * `*_join` member functions
+ *
+ * @note Behavior is undefined if the build table contains duplicates.
+ * @note All NaNs are considered as equal
+ *
+ * @tparam HasNested Flag indicating whether there are nested columns in build/probe table
+ */
+// TODO: `HasNested` to be removed via dispatching
+template <has_nested HasNested = has_nested::NO>
+class unique_hash_join {
```

Comment:
We've hashed this out a few times on the nomenclature between `unique` and `distinct`. I suggest using `distinct` to avoid confusion with the different semantics of `std::unique`.

Reply:
Thanks for catching this.

I believe I was involved in most of the naming discussions you mentioned but still didn't question the misuse of `unique`. All changed to `distinct_*` as requested.